java/vector/src/main/java/org/apache/arrow/vector/TimeStampMicroVector.java

```diff
@@ -110,12 +113,8 @@ public LocalDateTime getObject(int index) {
     if (isSet(index) == 0) {
       return null;
     } else {
-      /* value is truncated when converting microseconds to milliseconds in order to use DateTime type */
       final long micros = valueBuffer.getLong(index * TYPE_WIDTH);
-      final long millis = java.util.concurrent.TimeUnit.MICROSECONDS.toMillis(micros);
-      final org.joda.time.LocalDateTime localDateTime = new org.joda.time.LocalDateTime(millis,
-              org.joda.time.DateTimeZone.UTC);
-      return localDateTime;
+      return DateUtility.getLocalDateTimeFromEpochMicro(micros);
```

Comment:
This change here is a change in API. It will break all applications using this getObject API.